var/spack/repos/builtin/packages/rust/package.py

```diff
@@ -541,13 +523,35 @@ def configure(self, spec, prefix):
             )
             )
 
-    def build(self, spec, prefix):
-        python('./x.py', 'build', extra_env={
+    def install(self, spec, prefix):
+        jobs = spack.config.get('config:build_jobs') if self.parallel else 1
+
+        rustflags = []
+
+        target_cpu = \
+            RustQuery(
+                Executable(
+                    join_path(
+                        self.stage.source_path,
+                        'spack_bootstrap/bin/rustc'
+                    )
+                )
+            ).target_cpu(self.spec)
+
+        if target_cpu:
+            rustflags.extend(["-C", "target-cpu={0}".format(target_cpu)])
+
+        python('./x.py', 'install', '-j', str(jobs), extra_env={
             # vendored libgit2 wasn't correctly building (couldn't find the
             # vendored libssh2), so let's just have spack build it
             'LIBSSH2_SYS_USE_PKG_CONFIG': '1',
-            'LIBGIT2_SYS_USE_PKG_CONFIG': '1'
+            'LIBGIT2_SYS_USE_PKG_CONFIG': '1',
+            'RUSTFLAGS': " ".join(rustflags)
         })
 
-    def install(self, spec, prefix):
-        python('./x.py', 'install')
+    def setup_dependent_package(self, module, dependent_spec):
+        """Called before cargo modules' build() methods."""
+
+        module.cargo = Executable(join_path(self.prefix.bin, 'cargo'))
```

Comment:
This makes it so dependent packages that are using cargo directly will supply the `--locked` and `--offline` flags to help enforce the use of Spack's mechanisms for caching Cargo dependencies.